runtime/compiler/z/codegen/J9BCDTreeEvaluator.cpp

```diff
@@ -2070,10 +2070,36 @@ J9::Z::TreeEvaluator::BCDCHKEvaluatorImpl(TR::Node * node,
       callNode->setAndIncChild(i, childRootNode->getChild(i + callChildStartIndex));
 
    // Evaluate secondChild's children, if the secondChild is an address node into a byte[]
-   if(isResultPD && secondChild->getNumChildren() == 2 && secondChild->getReferenceCount() > 1)
-      {
-      cg->evaluate(secondChild->getFirstChild());
-      cg->evaluate(secondChild->getSecondChild());
+   if(isResultPD && secondChild->getNumChildren() == 2)
```

Comment:
Would secondChild not always be `aladd` node ? I personally found it bit confusing just code-readability point of view, and would have chose different name. Also would have checked the reference count of the children before evaluating it (If they are no longer referenced after this node, would still be OK to delay the evaluation in OOL I think. 

Given the change is blocking a functional issue, I am OK with it.

Reply:
Sorry, completely, missed this comment. With my latest verification run, the `aladd` can be replaced by a `aloadi` if the offset is 0. This would only happen in off-heap mode.